File: Core Java/account-app/src/com/techlabs/account/Account.java
package com.techlabs.account;

public class Account {

	private final String name;
	private final int id;
	private double balance;
	public static int count;

	static{
		System.out.println("Inside static block....");
	}
	public Account(String name, int id, double balance) {
		System.out.println("Inside Constustor");
		this.name = name;
		this.id = id;
		this.balance = balance;
		count++;
	}

	public Account(String name, int id) {
		this(name, id, 500);
	}

	public void deposit(double balance) {
		this.balance = this.balance + balance;
	}

	public void withdraw(double balance) {
		if (this.balance >= 500 && this.balance - balance >= 500) {
			this.balance = this.balance - balance;
		} else {
			throw new RuntimeException(
					"Your balance is low Minimum balance require 500");
		}
	}

	public int getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public double getBalance() {
		return balance;
	}

	public int getCount() {
		return count;
	}

	public int getAccountNumber() {
		return getCount();
	}
	
	public static int getAccountHolder()
	{
		return count;
	}
}

File: DOTNET/UNITTEST/AccountUnitTestSolution/AccountCore/Account.cs
using System;

namespace AccountApp
{
    public class Account
    {
        private readonly String _name;
        private readonly int _id;
        private double _balance;
        private static int count = 0;
        private static readonly int MINIMUM_BALANCE = 500;

        public Account(String name, int id, double balance)
        {
            _name = name;
            _id = id;
            _balance = balance;
            count++;
        }

        public String Name
        {
            get
            {
                return _name;
            }
        }

        public int Id
        {
            get
            {
                return _id;
            }
        }

        public double Balance
        {
            get
            {
                return _balance;
            }
        }

        public void Deposit(double balance)
        {
            _balance = _balance + balance;
        }

        public void Withdraw(double balance)
        {
            if (_balance >= MINIMUM_BALANCE && _balance - balance > MINIMUM_BALANCE)
            {
                _balance = _balance - balance;
            }
            else
            {
                throw new Exception
                    ("Your balance is low,Minimum require balance is 500.");
            }
        }

        public static int GetNumberOfAccount()
        {
            return count;
        }

    }
}